### PR TITLE
scd2html: init at 1.0.0

### PR DESCRIPTION
**Pull request by ianmjones**
## Description of changes

Add scd2html - generates HTML from scdoc source files.

Home page:- https://git.sr.ht/~bitfehler/scd2html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ianmjones** [comment]
Result of `nixpkgs-review pr 259582` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scd2html</li>
  </ul>
</details>

File: pkgs/by-name/sc/scd2html/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    scdoc
+  ];
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace "-static" ""
```
Comment (pbsds) [inline pkgs/by-name/sc/scd2html/package.nix:27]:
```suggestion
      --replace "LDFLAGS+=-static" ""
```

File: pkgs/by-name/sc/scd2html/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  '';
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+  ];
+
+  doCheck = true;
```
Comment (pbsds) [inline pkgs/by-name/sc/scd2html/package.nix:34]: From logs:

```
running tests
check flags: SHELL=/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash PREFIX=/nix/store/a5xry101c2qbsg4z567x40d3kf3hyygv-scd2html-1.0.0 VERBOSE=y check
make: Nothing to be done for 'check'.
```

This check phase is inexpensive, so we could leave it for the sake of future proofing, but it advertises the wrong behaviour. I would appreciate a comment

File: pkgs/by-name/sc/scd2html/package.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  meta = with lib; {
+    description = "scd2html generates HTML from scdoc source files";
+    homepage = "https://git.sr.ht/~bitfehler/scd2html";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ianmjones ];
+    platforms = platforms.unix;
+  };
```
Comment (pbsds) [inline pkgs/by-name/sc/scd2html/package.nix:42]:
```suggestion
    mainProgram = "scd2html";
  };
```

File: pkgs/by-name/sc/scd2html/package.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, fetchFromSourcehut
+, buildPackages
```
Comment (pbsds) [inline pkgs/by-name/sc/scd2html/package.nix:4]:
```suggestion
```

File: pkgs/by-name/sc/scd2html/package.nix (hunk 2 above)
Comment (onemoresuza) [inline pkgs/by-name/sc/scd2html/package.nix:34]: I've built it with `enableParallelBuinding = true` and it succeeded, so it seems there's no harm in enabling it.

```suggestion

  enableParallelBuinding = true;

  doCheck = true;
```

**Review by pbsds — CHANGES_REQUESTED** [review]
Looking good, some nits

File: pkgs/by-name/sc/scd2html/package.nix (hunk 4 above)
Comment (onemoresuza) [inline pkgs/by-name/sc/scd2html/package.nix:4]: The `buildPackages` argument it's not being used. Could you remove it?

```suggestion

```

File: pkgs/by-name/sc/scd2html/package.nix
Comment (onemoresuza) [inline pkgs/by-name/sc/scd2html/package.nix:1]: `ofborg` is throwing an error when trying to build it on [x86_64-darwin](https://logs.ofborg.org/?attempt_id=3f6d4a3f-41de-4c1e-8643-90bf0437df84&key=nixos%2Fnixpkgs.259582).

It seems to be related `fmemopen` not being available, as the program is right now, to such system.

File: pkgs/by-name/sc/scd2html/package.nix (hunk 2 above)
Comment (onemoresuza) [reply pkgs/by-name/sc/scd2html/package.nix:34]: Updated it. Thanks for catching it, @pbsds.

**Comment by ianmjones** [comment]
Thanks for the reviews @pbsds & @onemoresuza, all fixed up.

**Comment by pbsds** [comment]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

---

Result of `nixpkgs-review pr 259582` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scd2html</li>
  </ul>
</details>

**Review by pbsds — APPROVED** [review]

**Review by delroth — APPROVED** [review]
Unsure why this was sent my way, but pbsds did such a thorough job reviewing this anyway that I don't mind :-)

**Pull request merged**